server/src/main/java/io/pravega/schemaregistry/server/rest/resources/GroupResourceImpl.java

```diff
@@ -0,0 +1,708 @@
+import io.pravega.schemaregistry.contract.generated.rest.model.CreateGroupRequest;
+import io.pravega.schemaregistry.contract.generated.rest.model.EncodingId;
+import io.pravega.schemaregistry.contract.generated.rest.model.EncodingInfo;
+import io.pravega.schemaregistry.contract.generated.rest.model.GetEncodingIdRequest;
+import io.pravega.schemaregistry.contract.generated.rest.model.GroupHistory;
+import io.pravega.schemaregistry.contract.generated.rest.model.ListGroupsResponse;
+import io.pravega.schemaregistry.contract.generated.rest.model.SchemaInfo;
+import io.pravega.schemaregistry.contract.generated.rest.model.SchemaVersionsList;
+import io.pravega.schemaregistry.contract.generated.rest.model.SchemaWithVersion;
+import io.pravega.schemaregistry.contract.generated.rest.model.UpdateCompatibilityRequest;
+import io.pravega.schemaregistry.contract.generated.rest.model.Valid;
+import io.pravega.schemaregistry.contract.generated.rest.model.ValidateRequest;
+import io.pravega.schemaregistry.contract.generated.rest.model.VersionInfo;
+import io.pravega.schemaregistry.contract.transform.ModelHelper;
+import io.pravega.schemaregistry.contract.v1.ApiV1;
+import io.pravega.schemaregistry.exceptions.CodecTypeNotRegisteredException;
+import io.pravega.schemaregistry.exceptions.IncompatibleSchemaException;
+import io.pravega.schemaregistry.exceptions.PreconditionFailedException;
+import io.pravega.schemaregistry.exceptions.SerializationFormatMismatchException;
+import io.pravega.schemaregistry.server.rest.ServiceConfig;
+import io.pravega.schemaregistry.server.rest.auth.AuthHandlerManager;
+import io.pravega.schemaregistry.service.SchemaRegistryService;
+import io.pravega.schemaregistry.storage.ContinuationToken;
+import io.pravega.schemaregistry.storage.StoreExceptions;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.core.Response;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static io.pravega.auth.AuthHandler.Permissions.READ;
+import static io.pravega.auth.AuthHandler.Permissions.READ_UPDATE;
+import static javax.ws.rs.core.Response.Status;
+
+/**
+ * Schema Registry Resource implementation.
+ */
+@Slf4j
+public class GroupResourceImpl extends AbstractResource implements ApiV1.GroupsApiAsync {
+    private static final int DEFAULT_LIST_GROUPS_LIMIT = 100;
+    
+    public GroupResourceImpl(SchemaRegistryService registryService, ServiceConfig config, Executor executor) {
+        super(registryService, config, executor);
+    }
+
+    @Override
+    public void listGroups(String namespace, String continuationToken, Integer limit, 
+                           AsyncResponse asyncResponse) {
+        log.info("List Groups called");
```

Comment:
If we are keeping this log statement, it'd be useful to add the context too like namespace, token and limit, here and in other places we are tracing the request. 

My worry with info level logging is the flooding of logs.